cql3/expr/to_restriction.cc

```diff
@@ -342,8 +342,11 @@ ::shared_ptr<restrictions::restriction> to_restriction(const expression& e,
         throw exceptions::invalid_request_exception(format("Unsupported \"!=\" relation: {}", prepared_binop));
     }
 
-    if (prepared_binop.op == oper_t::IS_NOT && !is<null>(prepared_binop.rhs)) {
-        throw exceptions::invalid_request_exception(format("Unsupported \"IS NOT\" relation: {}", prepared_binop));
+    if (prepared_binop.op == oper_t::IS_NOT) {
```

Comment:
If we had a bug here, it probably means our tests do not cover this case :-( i.e., that "a IS NOT 3" should be refused when creating a materialized view - not cause a weird failure later when adding rows.
Would be good to add a regression test when we find such bugs.

Reply:
Added a test, although it turns out that `IS NOT NULL` is checked earlier at the grammar level.